test/src/features/instanceID.cpp

```diff
@@ -0,0 +1,70 @@
+#include "lidars.hpp"
+#include "scenes.hpp"
+#include <scene/Scene.hpp>
+#include "utils.hpp"
+#ifdef RGL_BUILD_ROS2_EXTENSION
+#include <rgl/api/extensions/ros2.h>
+#endif
+
+class InstanceIDTest : public RGLTest{};
+
+TEST_F(InstanceIDTest, BaseTest)
+{
+    setupThreeBoxScene(nullptr);
+
+    rgl_node_t useRaysNode = nullptr, raytraceNode = nullptr, compactNode = nullptr, formatNode = nullptr;
+    std::vector<rgl_mat3x4f> rays = makeLidar3dRays(360, 180, 0.72, 0.36);
+
+    std::vector<rgl_field_t> formatFields = {
+        XYZ_F32,
+        ENTITY_IDX_I32
+    };
+    struct FormatStruct
+    {
+        Field<XYZ_F32>::type xyz;
+        Field<ENTITY_IDX_I32>::type entityId;
+    } formatStruct;
```

Comment:
You don't need this. The same effect (and less typing) can be achieved using YieldPointsNode (and getting results as SoAs).